src/Microsoft.SqlTools.ServiceLayer/AzureBlob/SharedAccessSignatureCreator.cs

```diff
@@ -0,0 +1,115 @@
+    {
+        private Server sqlServer;
+
+        public SharedAccessSignatureCreator(Server sqlServer)
+        {
+            this.sqlServer = sqlServer;
+        }
+
+        public string CreateSqlSASCredential(string accountName, string accountKey, string containerUri, string expirationDateString)
+        {
+            DateTimeOffset? expirationDate = null;
+            if (!String.IsNullOrEmpty(expirationDateString))
+            {
+                expirationDate = DateTimeOffset.Parse(expirationDateString, CultureInfo.InvariantCulture);
+            }
+            var containerClient = new BlobContainerClient(new Uri(containerUri), new StorageSharedKeyCredential(accountName, accountKey));
+            Uri secretStringUri = GetServiceSasUriForContainer(containerClient, null, expirationDate);
+            string secretString = secretStringUri.ToString().Split('?')[1];
+            string identity = "Shared Access Signature";
+            WriteSASCredentialToSqlServer(containerUri, identity, secretString);
+            return secretString;
+        }
+
+        /// <summary>
+        /// Create sql sas credential with the given credential name
+        /// </summary>
+        /// <param name="credentialName">Name of sas credential, here is the same of the full container url.</param>
+        /// <param name="identity">Identity for credential, here is fixed as "Shared Access Signature"</param>
+        /// <param name="secretString">Secret of credential, which is sharedAccessSignatureForContainer </param>
+        /// <returns> The newly created SAS credential</returns>
+        public Credential WriteSASCredentialToSqlServer(string credentialName, string identity, string secretString)
+        {
+            // Format of Sql SAS credential: 
+            // CREATE CREDENTIAL [https://<StorageAccountName>.blob.core.windows.net/<ContainerName>] WITH IDENTITY = N'Shared Access Signature', 
+            // SECRET = N'sv=2014-02-14&sr=c&sig=lxb2aXr%2Bi0Aeygg%2B0a4REZ%2BqsUxxxxxxsqUybg0tVzg%3D&st=2015-10-15T08%3A00%3A00Z&se=2015-11-15T08%3A00%3A00Z&sp=rwdl'
+            //
+            CredentialCollection credentials = sqlServer.Credentials;
+
+            Credential azureCredential = new Credential(sqlServer, credentialName);
+
+            // Container can have many SAS credentials coexisting, here we'll always drop existing one once customer choose to create new credential 
+            // since sql customer has no way to know its existency and even harder to retrive its secret string. 
+            if (credentials.Contains(credentialName))
+            {
+                Credential oldCredential = credentials[credentialName];
+                oldCredential.Drop();
+            }
+
+            try
+            {
+                azureCredential.Create(identity, secretString);
+            }
+            catch (Exception ex)
+            {
+                throw new FailedOperationException("Create credential failed.", ex);
```

Comment:
I'd consider having this be wrapped around the entire method since there's a lot of places that an exception can be thrown - and it'd be nice for all of those to know that the error was during the process of writing the SAS to the server. (you could also throw a separate exception for each place - either is fine)